server/stream.go

```diff
@@ -44,3 +44,22 @@ func (s *server) invokeSendEndpoint() endpoint.SendEndpoint {
 		return stream.SendMsg(req)
 	}
 }
+
+// contextStream is responsible for solving ctx diverge in server side streaming.
+// it receives the ctx from previous middlewares and the Stream that exposed to users，then rewrite
+// Context() method so that users could call Stream.Context() in handler to get the processed ctx.
+type contextStream struct {
+	streaming.Stream
+	ctx context.Context
+}
+
+func (cs *contextStream) Context() context.Context {
```

Comment:
the allocation can be eliminated without pointer receiver. Please make sure every implementation is zero heap allocation in hot paths.

Reply:
Got it. I'll check and modify this part.